snuba/datasets/errors_replacer.py

```diff
@@ -373,17 +375,30 @@ def process_delete_groups(
 
 
 def process_tombstone_events(
-    message: Mapping[str, Any], required_columns: Sequence[str]
+    message: Mapping[str, Any],
+    required_columns: Sequence[str],
+    state_name: ReplacerState,
 ) -> Optional[Replacement]:
     event_ids = message["event_ids"]
     if not event_ids:
         return None
 
+    old_primary_hash = message.get("old_primary_hash")
+
+    if old_primary_hash and state_name == ReplacerState.EVENTS:
+        # old_primary_hash flag means the event is only tombstoned
+        # because it will be reinserted with a changed primary_hash. Since
+        # primary_hash is part of the sortkey/primarykey in the ERRORS table,
+        # we need to tombstone the old event. In the old EVENTS table we do
+        # not.
+        return None
+
     # XXX: We need to construct a query that works on both event_id columns,
     # either represented as UUID or as hyphenless FixedString. That's why we
     # use replaceAll(toString()).
     where = """\
         PREWHERE replaceAll(toString(event_id), '-', '') IN (%(event_ids)s)
+        AND (%(old_primary_hash)s IS NULL OR primary_hash = %(old_primary_hash)s)
```

Comment:
What's the reason for adding the extra condition at the start `%(old_primary_hash)s IS NULL`?

Isn't `AND (%(old_primary_hash)s IS NULL OR primary_hash = %(old_primary_hash)s)` just the same as `AND primary_hash = %(old_primary_hash)s`

Reply:
if `old_primary_hash` is `null` we don't want to filter the events at all. I could not come up with an elegant way to conditionally add this clause, so instead the thing now emits `AND (NULL IS NULL OR ...)` which is always true.